demo/admin/src/products/future/generated/ProductForm.tsx

```diff
@@ -97,6 +105,11 @@ export function ProductForm({ id }: FormProps): React.ReactElement {
         if (await saveConflict.checkForConflicts()) throw new Error("Conflicts detected");
         const output = {
             ...formValues,
+            packageDimensions: {
+                height: parseFloat(formValues.packageDimensions.height),
+                width: parseFloat(formValues.packageDimensions.width),
+                depth: parseFloat(formValues.packageDimensions.depth),
+            },
```

Comment:
This would probably cause an error (or a zero value) if a user doesn't enter a value in one of the fields.

Also, this is different to the initial values implementation: Here a single field is send if it's set in the form. In the initial values, the fields are only set if all three fields are set (`data.product.packageDimensions` is not null).

Reply:
event height: String(data.product.packageDimensions.height) would not be correct if height is optional, as String(undefined) => "undefined"

Reply:
true, undefined handling of numbet fields is currently broken and should be fixed in a different PR. As a first step a fix should be proposed for the handmade products demo.

Reply:
I looked into it in detail and it's halve broken. initialValues contains `myNumberField: "null"`, in output it's `myNumberField: NaN` but it's send as `myNumberField: null` in the graphql request. so in the end it works.

<img width="681" alt="image" src="https://github.com/vivid-planet/comet/assets/1324250/a83c2933-f2cd-4e3f-929d-e7de533d63e1">

Reply:
> As a first step a fix should be proposed for the handmade products demo.

packageDimensions and dimensions are optional, but width, height and depth aren't. How should this be handled in the handmade products demo? just define width, height and depth as required and therefore make packageDimensions required when created/edited via form. (this is the current behavior with the generated code) or validate width, height and depth as required if at least one of them contains a value and else omit the whole packageDimensions?

or should we nifty handle this situation by wrapping them in some kind of checkable/collapsable fieldset? can we force any optional nested structure into a fieldset?

Reply:
> packageDimensions and dimensions are optional, but width, height and depth aren't.

I think for packageDimensions it's currently broken: it's an embeddable where the nested fields are created as non-nullable columns in the database, so packageDimensions actually can't be null. That should be changed. (own PR!)

dimensions is different, that is json so it can be null. The correct UI would be a checkable fieldset. But that is something for later, let's ignore that field in admin (handmade+generator) for now.

Reply:
> I looked into it in detail and it's halve broken. initialValues contains myNumberField: "null"

what? `"null"?` that is so wrong :)

Reply:
> I think for packageDimensions it's currently broken: it's an embeddable where the nested fields are created as non-nullable columns in the database, so packageDimensions actually can't be null. That should be changed. (own PR!)

done in: https://github.com/vivid-planet/comet/pull/1730

Reply:
> ... let's ignore that field in admin (handmade+generator) for now.

so I won't add `| null` for optional objects in path.

but there is still the problem with optional number-fields. Should I implement this two times, as you mentioned "undefined handling of numbet fields is currently broken and should be fixed in a different PR"? because the number-fields code in current future-admin generator is rewritten/moved in this pull-request.

Reply:
> so I won't add | null for optional objects in path.

:+1:, something for a follow up PR

> but there is still the problem with optional number-fields. Should I implement this two times, as you mentioned "undefined handling of numbet fields is currently broken and should be fixed in a different PR"? because the number-fields code in current future-admin generator is rewritten/moved in this pull-request.

yes, please